### PR TITLE
Adds DropLocation(). To generally be preferred to get_turf()

### LINKED ISSUE #29802: tablet has a print feature, if you print something inside of a locker it dispenses the printed paper outside the locker

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #29802 

Unlike `get_turf()`this will sensibly pick if `loc` can be safely used

### CODE REVIEW

**Comment by optimumtact** [comment]
your procs look ugly and are offensive with the Upper Case everywhere

I hope you are run over by a small fluffy dog

**Comment by Cyberboss** [comment]
I would appreciate that

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -700,3 +700,6 @@
 	if(anchored || throwing)
 		return FALSE
 	return TRUE
+
+/atom/movable/DropLocation(atom/movable/dropee = src)
+	return ..()
```
Comment (MrPerson) [inline code/game/atoms_movable.dm]: This seems idiotic. Just define the arg on the parent and be done with it.

Comment (Cyberboss) [inline code/game/atoms_movable.dm]: relic of a different implementation

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -477,3 +477,9 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 	valid_territory = FALSE
 	blob_allowed = FALSE
 	addSorted()
+
+/area/AllowDrop()
+	CRASH("Ya dun fuked up")
```
Comment (optimumtact) [inline code/game/area/areas.dm]: can you change these to be more descriptive

**Review by lzimann — DISMISSED** [review]
what oranges said

**Comment by optimumtact** [comment]
please rename it to drop_location :pray:

**Comment by optimumtact** [comment]
Java/C# "enterprise" devs out in full force here I see

**Comment by Cyberboss** [comment]
Enterprise pays bill son

**Comment by lzimann** [comment]
what's the point of paying bills if it looks ugly

**Comment by Cyberboss** [comment]
I beg to differ, jimbo

**Comment by RemieRichards** [comment]
revert 4c82734

**Comment by optimumtact** [comment]
my eyes

**Comment by optimumtact** [comment]
revert efd83ee

**Comment by RemieRichards** [comment]
law 0 fruits aren't human

**Pull request merged**